pkg/collector/corechecks/systemd/systemd_test.go

```diff
@@ -466,6 +467,60 @@ unit_names:
 	mockSender.AssertNumberOfCalls(t, "ServiceCheck", 4)
 }
 
+// When a value is not set (`[Not set]` when running `systemctl show my.service`), dbus returns MaxUint64
+func TestMetricValuesNotSet(t *testing.T) {
+	rawInstanceConfig := []byte(`
+unit_names:
+ - unit1.service
+`)
+
+	stats := createDefaultMockSystemdStats()
+	stats.On("ListUnits", mock.Anything).Return([]dbus.UnitStatus{
+		{Name: "unit1.service", ActiveState: "active", LoadState: "loaded"},
+	}, nil)
+	stats.On("UnixNow").Return(int64(1000))
+	stats.On("GetUnitTypeProperties", mock.Anything, "unit1.service", dbusTypeMap[typeService]).Return(getCreatePropertieWithDefaults(map[string]interface{}{
+		"CPUUsageNSec":  uint64(10),
+		"MemoryCurrent": uint64(math.MaxUint64),
+		"TasksCurrent":  uint64(30),
+		"NRestarts":     uint64(40),
+	}), nil)
+	stats.On("GetUnitTypeProperties", mock.Anything, "unit1.service", dbusTypeMap[typeUnit]).Return(map[string]interface{}{
+		"ActiveEnterTimestamp": uint64(100 * 1000 * 1000),
+	}, nil)
+
+	stats.On("GetVersion", mock.Anything).Return(systemdVersion)
+
+	check := SystemdCheck{stats: stats}
+	senderManager := mocksender.CreateDefaultDemultiplexer()
+	check.Configure(senderManager, integration.FakeConfigHash, rawInstanceConfig, nil, "test")
+
+	// setup expectation
+	mockSender := mocksender.NewMockSenderWithSenderManager(check.ID(), senderManager)
+	mockSender.On("Gauge", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
+	mockSender.On("ServiceCheck", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return()
+	mockSender.On("Commit").Return()
+
+	// run
+	check.Run()
+
+	// assertions
+	tags := []string{"unit:unit1.service"}
+	mockSender.AssertCalled(t, "Gauge", "systemd.unit.uptime", float64(900), "", tags)
+	mockSender.AssertCalled(t, "Gauge", "systemd.unit.monitored", float64(1), "", tags)
+	mockSender.AssertCalled(t, "Gauge", "systemd.unit.active", float64(1), "", tags)
+	mockSender.AssertCalled(t, "Gauge", "systemd.unit.loaded", float64(1), "", tags)
+	mockSender.AssertCalled(t, "Gauge", "systemd.service.cpu_time_consumed", float64(10), "", tags)
+	mockSender.AssertCalled(t, "Gauge", "systemd.service.task_count", float64(30), "", tags)
+	mockSender.AssertCalled(t, "Gauge", "systemd.service.restart_count", float64(40), "", tags)
```

Comment:
Would it be possible to assert that we **didn't** submit something? i.e. this "MemoryCurrent" thing.

Reply:
That's what we indirectly do [here](https://github.com/DataDog/datadog-agent/pull/23436/files#diff-ac391a17448b7693a18a772a8baf4e3963eacaefdb32f7e3eb2390027f630dbaR519) in fact

Reply:
I'll add a AssertNotCalled for clarity

Reply:
Thanks! I think the AssertNotCalled adds clarity!